src/core/Group.cpp

```diff
@@ -677,6 +677,11 @@ Entry* Group::findEntryByPathRecursive(const QString& entryPath, const QString&
             || (!entryPath.startsWith("/") && entry->title() == entryPath)) {
             return entry;
         }
+        bool found = false;
+        found = entry->uuid() == QUuid::fromRfc4122(QByteArray::fromHex(entry->uuid().toString().toLatin1()));
+        if (found) {
+            return entry;
+        }
         // clang-format on
```

Comment:
I don't understand what you are trying to accomplish here. `entry->uuid() == QUuid::fromRfc4122(QByteArray::fromHex(entry->uuid().toString().toLatin1()))` is always true and an entirely unnecessary back-and-forth chain of conversions.
The loop now exits with an arbitrary entry unless you get very lucky and the entry you are actually looking for is the very first one in the list.

Reply:
Grabbed this code from another location and missed this.
Ran another test and see that it is grabbed the 1st entry. Should've caught this :man_facepalming: .

Am matching the entry string correctly now with QUuid::Id128, as this removes the braces and dashes. Tested with multiple entries.